lib/sourceror/zipper.ex

```diff
@@ -81,12 +84,22 @@ defmodule Sourceror.Zipper do
   def zip(node), do: new(node)
 
   @doc """
-  Walks the `zipper` all the way up and returns the top `zipper`.
+  Walks the `zipper` to the top of the current subtree and returns that `zipper`.
   """
   @spec top(t) :: t
   def top(%Z{path: nil} = zipper), do: zipper
   def top(zipper), do: zipper |> up() |> top()
 
+  @doc """
+  Walks the `zipper` all the way up, breaking out of any subtrees and returns the top-most `zipper`.
+  """
+  @spec all_the_way_up(t) :: t
+  def all_the_way_up(%Z{supertree: supertree} = zipper) when not is_nil(supertree) do
+    all_the_way_up(into(top(zipper), supertree))
+  end
+
+  def all_the_way_up(zipper), do: top(zipper)
```

Comment:
Maybe we can call this `topmost`? It feels a bit more consistent with the names `rightmost` and `leftmost`

Reply:
I like `topmost` 👍